alacritty/src/window.rs

```diff
@@ -135,13 +146,20 @@ fn create_gl_window(
     Ok(windowed_context)
 }
 
-/// A window which can be used for displaying the terminal
+/// A window which can be used for displaying the terminal.
 ///
-/// Wraps the underlying windowing library to provide a stable API in Alacritty
+/// Wraps the underlying windowing library to provide a stable API in Alacritty.
 pub struct Window {
     windowed_context: WindowedContext<PossiblyCurrent>,
     current_mouse_cursor: CursorIcon,
     mouse_visible: bool,
+    /// Wayland setting to control redrawing of the window, according to the frame callbacks.
+    #[cfg(not(any(target_os = "macos", windows)))]
+    pub should_draw: Arc<AtomicBool>,
+
+    /// Attached Wayland surface to create new frame events.
```

Comment:
Is 'create' new frame events the correct term here? Wouldn't `handle` be better? We're not only using it to create them, right? They also come back 'from' that surface.

Reply:
you're requesting a frame event and run a callback when server send you it back on dispatch_pending on event queue. Not sure about handle because of that.